From: gmoniker <gmoniker@gmail.com>
Date: Fri, 20 Mar 2015 12:13:05 +0100
Subject: PHP SegFault zend_hash_find (PHP#68486)

---
 sapi/apache2handler/sapi_apache2.c | 1 +
 1 file changed, 1 insertion(+)

--- php5.orig/sapi/apache2handler/sapi_apache2.c
+++ php5/sapi/apache2handler/sapi_apache2.c
@@ -688,6 +688,7 @@ zend_first_try {
 } zend_end_try();
 		}
 		apr_brigade_cleanup(brigade);
+		apr_pool_cleanup_run(r->pool, (void *)&SG(server_context), php_server_context_cleanup);
 	} else {
 		ctx->r = parent_req;
 	}
