--- mailgraph	2013-10-02 13:20:26.000000000 +0200
+++ mailgraph.postscreen	2013-10-02 13:18:58.000000000 +0200
@@ -659,6 +659,19 @@
 				event($time, 'rejected');
 			}
 		}
+                elsif($prog eq 'postscreen') {
+                        if($text =~ /NOQUEUE: reject: RCPT from .* /) {
+                                if($text =~ /Service unavailable; /) {
+					event($time, 'rejected');
+                                }
+                                elsif($text =~ /Protocol error; /) {
+					event($time, 'rejected');
+                                }
+                                elsif($text =~ /too many connections /) {
+					event($time, 'rejected');
+                                }
+                        }
+                }
 	}
 	elsif($prog eq 'sendmail' or $prog eq 'sm-mta') {
 		if($text =~ /\bmailer=local\b/ ) {
