--- a/TSRM/configure.in
+++ b/TSRM/configure.in
@@ -13,9 +13,6 @@ TSRM_BASIC_CHECKS
 TSRM_THREADS_CHECKS
 	
 AM_PROG_LIBTOOL
-if test "$enable_debug" != "yes"; then
-  AM_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl TSRM_PTHREAD
 
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -806,13 +806,13 @@ dnl PHP_BUILD_PROGRAM
 dnl
 AC_DEFUN([PHP_BUILD_PROGRAM],[
   OVERALL_TARGET=[]ifelse($1,,php,$1)
-  php_c_pre='$(CC)'
+  php_c_pre='$(LIBTOOL) --mode=compile $(CC)'
   php_c_meta='$(COMMON_FLAGS) $(CFLAGS_CLEAN) $(EXTRA_CFLAGS)'
-  php_c_post=' && echo > $[@]'
-  php_cxx_pre='$(CXX)'
+  php_c_post=
+  php_cxx_pre='$(LIBTOOL) --mode=compile $(CXX)'
   php_cxx_meta='$(COMMON_FLAGS) $(CXXFLAGS_CLEAN) $(EXTRA_CXXFLAGS)'
-  php_cxx_post=' && echo > $[@]'
-  php_lo=o
+  php_cxx_post=
+  php_lo=lo
 
   case $with_pic in
     yes) pic_setting='-prefer-pic';;
--- a/configure.in
+++ b/configure.in
@@ -1221,9 +1221,6 @@ PHP_CONFIGURE_PART(Configuring libtool)
 
 LDFLAGS="$LDFLAGS $PHP_AIX_LDFLAGS"
 AC_PROG_LIBTOOL
-if test "$enable_debug" != "yes"; then
-  PHP_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl libtool 1.4.3 needs this.
 PHP_SET_LIBTOOL_VARIABLE([--preserve-dup-deps])
