--- a/ext/readline/config.m4
+++ b/ext/readline/config.m4
@@ -12,7 +12,7 @@ fi
 
 if test "$PHP_READLINE" != "no"; then
   for i in $PHP_READLINE /usr/local /usr; do
-    test -f $i/include/readline/readline.h && READLINE_DIR=$i && break
+    test -f $i/include/editline/readline.h && READLINE_DIR=$i && break
   done
 
   if test -z "$READLINE_DIR"; then
@@ -66,7 +66,7 @@ if test "$PHP_READLINE" != "no"; then
 elif test "$PHP_LIBEDIT" != "no"; then
 
   for i in $PHP_LIBEDIT /usr/local /usr; do
-    test -f $i/include/readline/readline.h && LIBEDIT_DIR=$i && break
+    test -f $i/include/editline/readline.h && LIBEDIT_DIR=$i && break
   done
 
   if test -z "$LIBEDIT_DIR"; then
--- a/ext/readline/readline.c
+++ b/ext/readline/readline.c
@@ -29,9 +29,9 @@
 
 #if HAVE_LIBREADLINE || HAVE_LIBEDIT
 
-#include <readline/readline.h>
+#include <editline/readline.h>
 #ifndef HAVE_LIBEDIT
-#include <readline/history.h>
+#include <editline/history.h>
 #endif
 
 PHP_FUNCTION(readline);
