--- a/build/build2.mk
+++ b/build/build2.mk
@@ -52,6 +52,11 @@ $(TOUCH_FILES):
 
 aclocal.m4: configure.in acinclude.m4
 	@echo rebuilding $@
+	@libtoolize=`./build/shtool path glibtoolize libtoolize`; \
+	$$libtoolize --copy --automake --force; \
+	ltpath=`dirname $$libtoolize`; \
+	ltfile=`cd $$ltpath/../share/aclocal; pwd`/libtool.m4; \
+	cp $$ltfile ./build/libtool.m4
 	cat acinclude.m4 ./build/libtool.m4 > $@
 
 configure: aclocal.m4 configure.in $(config_m4_files)
