--- a/ext/standard/info.c
+++ b/ext/standard/info.c
@@ -446,7 +446,7 @@ PHPAPI void php_print_info(int flag TSRM
 		php_info_print_table_start();
 		php_info_print_table_row(2, "System", php_uname );
 		php_info_print_table_row(2, "Build Date", __DATE__ " " __TIME__ );
-#ifdef CONFIGURE_COMMAND
+#if 0
 		php_info_print_table_row(2, "Configure Command", CONFIGURE_COMMAND );
 #endif
 		if (sapi_module.pretty_name) {
