--- a/ext/mssql/config.m4
+++ b/ext/mssql/config.m4
@@ -9,12 +9,15 @@ PHP_ARG_WITH(mssql,for MSSQL support via
 
 if test "$PHP_MSSQL" != "no"; then
 
-  FREETDS_INSTALLATION_DIR=""
   if test "$PHP_MSSQL" = "yes"; then
-
     for i in /usr/local /usr; do
-      if test -f $i/include/tds.h; then
+      if test -f $i/include/sybdb.h; then
+        FREETDS_INSTALLATION_DIR=$i
+        FREETDS_INCLUDE_DIR=$i/include
+        break
+      elif test -f $i/include/freetds/sybdb.h; then
         FREETDS_INSTALLATION_DIR=$i
+        FREETDS_INCLUDE_DIR=$i/include/freetds
         break
       fi
     done
@@ -25,8 +28,12 @@ if test "$PHP_MSSQL" != "no"; then
 
   elif test "$PHP_MSSQL" != "no"; then
 
-    if test -f $PHP_MSSQL/include/tds.h; then
+    if test -f $PHP_MSSQL/include/sybdb.h; then
+      FREETDS_INSTALLATION_DIR=$PHP_MSSQL
+      FREETDS_INCLUDE_DIR=$PHP_MSSQL/include
+    elif test -f $PHP_MSSQL/include/freetds/sybdb.h; then
       FREETDS_INSTALLATION_DIR=$PHP_MSSQL
+      FREETDS_INCLUDE_DIR=$PHP_MSSQL/include/freetds
     else
       AC_MSG_ERROR(Directory $PHP_MSSQL is not a FreeTDS installation directory)
     fi
@@ -36,11 +43,9 @@ if test "$PHP_MSSQL" != "no"; then
      AC_MSG_ERROR(Could not find $FREETDS_INSTALLATION_DIR/lib/libsybdb.[a|so])
   fi
 
-  MSSQL_INCDIR=$FREETDS_INSTALLATION_DIR/include
-  MSSQL_LIBDIR=$FREETDS_INSTALLATION_DIR/lib
+  PHP_ADD_INCLUDE($FREETDS_INCLUDE_DIR)
+  PHP_ADD_LIBRARY_WITH_PATH(sybdb, $FREETDS_INSTALLATION_DIR/$PHP_LIBDIR, MSSQL_SHARED_LIBADD)
 
-  PHP_ADD_INCLUDE($MSSQL_INCDIR)
-  PHP_ADD_LIBRARY_WITH_PATH(sybdb, $MSSQL_LIBDIR, MSSQL_SHARED_LIBADD)
   PHP_NEW_EXTENSION(mssql, php_mssql.c, $ext_shared)
   AC_CHECK_LIB(dnet_stub, dnet_addr,
      [ PHP_ADD_LIBRARY_WITH_PATH(dnet_stub,,MSSQL_SHARED_LIBADD)
