--- a/ext/ldap/config.m4
+++ b/ext/ldap/config.m4
@@ -112,9 +112,9 @@ if test "$PHP_LDAP" != "no"; then
 
   dnl Save original values
   _SAVE_CPPFLAGS=$CPPFLAGS
-  _SAVE_LDFLAGS=$LDFLAGS
-  LDFLAGS="$LDFLAGS $LDAP_SHARED_LIBADD"
+  _SAVE_LIBS=$LIBS
   CPPFLAGS="$CPPFLAGS -I$LDAP_INCDIR"
+  LIBS="$LIBS $LDAP_SHARED_LIBADD"
 
   dnl Check for 3 arg ldap_set_rebind_proc
   AC_CACHE_CHECK([for 3 arg ldap_set_rebind_proc], ac_cv_3arg_setrebindproc,
@@ -137,5 +137,5 @@ if test "$PHP_LDAP" != "no"; then
 
   dnl Restore original values
   CPPFLAGS=$_SAVE_CPPFLAGS
-  LDFLAGS=$_SAVE_LDFLAGS
+  LIBS=$_SAVE_LIBS
 fi 
