--- a/ext/mime_magic/mime_magic.c
+++ b/ext/mime_magic/mime_magic.c
@@ -503,7 +503,7 @@ static int is_valid_mimetype(char *p, in
 		} while (*(++p) != '/');
 		++p;
 		do {
-			if (!isalnum(*p) && (*p != '-') && (*p != '.') && !isspace(*p)) {
+			if (!isalnum(*p) && (*p != '-') && (*p != '.') && (*p != '+') && !isspace(*p)) {
 				return 0;
 			}
 		} while (*(++p));
@@ -636,6 +636,15 @@ static int parse(char *l, int lineno)
     else if (strncmp(l, "string", NSTRING) == 0) {
 		m->type = STRING;
 		l += NSTRING;
+		if (*l == '/') {
+			++l;
+			if ((*l == 'B') || (*l == 'b') || (*l == 'c')) {
+				++l;
+				if ((*l == 'B') || (*l == 'b') || (*l == 'c')) {
+					++l;
+				}
+			}
+		}
     }
     else if (strncmp(l, "date", NDATE) == 0) {
 		m->type = DATE;
@@ -729,7 +738,7 @@ static int parse(char *l, int lineno)
 	if (!is_valid_mimetype(l, strlen(l))) {
 		if(MIME_MAGIC_G(debug))
 			php_error_docref("http://www.php.net/mime_magic" TSRMLS_CC, E_WARNING, ": (%s:%d) '%s' is not a valid mimetype, entry skipped", MIME_MAGIC_G(magicfile), lineno, l);
-		return -1;
+		return 0;
 	}
 	
 	strlcpy(m->desc, l, sizeof(m->desc));
