--- php.orig/sapi/cli/php.1.in
+++ php/sapi/cli/php.1.in
@@ -1,6 +1,5 @@
 .TH PHP 1 "2009" "The PHP Group" "Scripting Language"
 .SH NAME
-.TP 15
 php \- PHP Command Line Interface 'CLI'
 .SH SYNOPSIS
 .B php
--- php.orig/scripts/man1/php-config.1.in
+++ php/scripts/man1/php-config.1.in
@@ -1,6 +1,5 @@
 .TH php\-config 1 "2006" "The PHP Group" "Scripting Language"
 .SH NAME
-.TP 15
 php\-config \- get information about PHP configuration and compile options
 .SH SYNOPSIS
 .B php\-config
--- php.orig/scripts/man1/phpize.1.in
+++ php/scripts/man1/phpize.1.in
@@ -1,7 +1,6 @@
 .TH phpize 1 "2006" "The PHP Group" "Scripting Language"
 .SH NAME
-.TP 15
-phpize - prepare a PHP extension for compiling
+phpize \- prepare a PHP extension for compiling
 .SH SYNOPSIS
 .B phpize
 [options]
