--- a/ext/mysql/config.m4
+++ b/ext/mysql/config.m4
@@ -86,7 +86,7 @@ if test "$PHP_MYSQL" != "no"; then
 Note that the MySQL client library is not bundled anymore!])
   fi
 
-  if test "$enable_maintainer_zts" = "yes"; then
+  if true || test "$enable_maintainer_zts" = "yes"; then
     MYSQL_LIBNAME=mysqlclient_r
   else
     MYSQL_LIBNAME=mysqlclient
--- a/ext/mysqli/config.m4
+++ b/ext/mysqli/config.m4
@@ -26,7 +26,7 @@ dnl  fi
   if test "$PHP_EMBEDDED_MYSQLI" = "yes"; then
     AC_DEFINE(HAVE_EMBEDDED_MYSQLI, 1, [embedded MySQL support enabled])
     MYSQL_LIB_CFG='--libmysqld-libs'
-  elif test "$enable_maintainer_zts" = "yes"; then
+  elif true || test "$enable_maintainer_zts" = "yes"; then
     MYSQL_LIB_CFG='--libs_r'
     MYSQL_LIB_NAME='mysqlclient_r'
   else
--- a/ext/pdo_mysql/config.m4
+++ b/ext/pdo_mysql/config.m4
@@ -58,7 +58,7 @@ if test "$PHP_PDO_MYSQL" != "no"; then
     if test "x$SED" = "x"; then
       AC_PATH_PROG(SED, sed)
     fi
-    if test "$enable_maintainer_zts" = "yes"; then
+    if true || test "$enable_maintainer_zts" = "yes"; then
       PDO_MYSQL_LIBNAME=mysqlclient_r
       PDO_MYSQL_LIBS=`$PDO_MYSQL_CONFIG --libs_r | $SED -e "s/'//g"`
     else
