--- php.orig/sapi/cli/php.1.in
+++ php/sapi/cli/php.1.in
@@ -69,7 +69,7 @@ specified by \-F to be executed.
 You can access the input line by \fB$argn\fP. While processing the input lines
 .B $argi 
 contains the number of the actual line being processed. Further more
-the paramters \-B and \-E can be used to execute 
+the parameters \-B and \-E can be used to execute 
 .IR code
 (see \-r) before and
 after all input lines have been processed respectively. Notice that the
@@ -316,7 +316,7 @@ The configuration file for the CGI versi
 The configuration file for the version of PHP that apache2 uses.
 .SH EXAMPLES
 .TP 5
-\fIphp -r 'echo "Hello World\\n";'\fP
+\fIphp \-r 'echo "Hello World\\n";'\fP
 This command simply writes the text "Hello World" to standard out.
 .TP
 \fIphp \-r 'print_r(gd_info());'\fP
@@ -340,7 +340,7 @@ configuration information. If you then c
 Using this PHP command you can count the lines being input.
 .TP
 \fIphp \-R '@$l+=count(file($argn));' \-E 'echo "Lines:$l\\n";'\fP
-In this example PHP expects each input line beeing a file. It counts all lines 
+In this example PHP expects each input line being a file. It counts all lines 
 of the files specified by each input line and shows the summarized result. 
 You may combine this with tools like find and change the php scriptlet.
 .TP
