Index: php-5.2.17/TSRM/configure.in
===================================================================
--- php-5.2.17.orig/TSRM/configure.in	2015-04-13 04:58:10.225313463 +0200
+++ php-5.2.17/TSRM/configure.in	2015-04-13 04:58:42.501312797 +0200
@@ -13,9 +13,6 @@
 TSRM_THREADS_CHECKS
 	
 AM_PROG_LIBTOOL
-if test "$enable_debug" != "yes"; then
-  AM_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl TSRM_PTHREAD
 
Index: php-5.2.17/configure.in
===================================================================
--- php-5.2.17.orig/configure.in	2015-04-13 04:58:10.225313463 +0200
+++ php-5.2.17/configure.in	2015-04-13 04:58:42.501312797 +0200
@@ -1335,9 +1335,6 @@
 ])
 AC_PROG_LIBTOOL
 
-if test "$enable_debug" != "yes"; then
-  PHP_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl libtool 1.4.3 needs this.
 PHP_SET_LIBTOOL_VARIABLE([--preserve-dup-deps])
