Index: php-5.2.17/ext/ldap/ldap.c
===================================================================
--- php-5.2.17.orig/ext/ldap/ldap.c	2015-04-13 04:58:10.049313467 +0200
+++ php-5.2.17/ext/ldap/ldap.c	2015-04-13 04:58:42.997312802 +0200
@@ -1342,7 +1342,7 @@
 	}
 
 	i=0;
-	while (ldap_value[i] != NULL) i++;
+	while (ldap_value && ldap_value[i] != NULL) i++;
 	count = i;
 
 	array_init(return_value);
@@ -1352,7 +1352,8 @@
 		add_index_string(return_value, i, ldap_value[i], 1);
 	}
 
-	ldap_value_free(ldap_value);
+	if (ldap_value)
+		ldap_value_free(ldap_value);
 }
 /* }}} */
 
