Index: php-5.2.17/acinclude.m4
===================================================================
--- php-5.2.17.orig/acinclude.m4	2015-04-13 04:58:42.753312805 +0200
+++ php-5.2.17/acinclude.m4	2015-04-13 04:58:43.449312792 +0200
@@ -1219,7 +1219,7 @@
     }
 
   ],[
-    ac_cv_pwrite=yes
+    ac_cv_pwrite=no
   ],[
     ac_cv_pwrite=no
   ],[
@@ -1248,7 +1248,7 @@
     exit(0);
     }
   ],[
-    ac_cv_pread=yes
+    ac_cv_pread=no
   ],[
     ac_cv_pread=no
   ],[
