Index: php-5.2.17/ext/standard/info.c
===================================================================
--- php-5.2.17.orig/ext/standard/info.c	2015-04-13 04:58:09.449313479 +0200
+++ php-5.2.17/ext/standard/info.c	2015-04-13 04:58:45.677312746 +0200
@@ -487,7 +487,7 @@
 		php_info_print_table_start();
 		php_info_print_table_row(2, "System", php_uname );
 		php_info_print_table_row(2, "Build Date", __DATE__ " " __TIME__ );
-#ifdef CONFIGURE_COMMAND
+#if 0
 		php_info_print_table_row(2, "Configure Command", CONFIGURE_COMMAND );
 #endif
 		if (sapi_module.pretty_name) {
Index: php-5.2.17/ext/standard/tests/general_functions/phpinfo.phpt
===================================================================
--- php-5.2.17.orig/ext/standard/tests/general_functions/phpinfo.phpt	2015-04-13 04:58:09.449313479 +0200
+++ php-5.2.17/ext/standard/tests/general_functions/phpinfo.phpt	2015-04-13 04:58:45.677312746 +0200
@@ -20,7 +20,6 @@
 
 System => %s
 Build Date => %s
-Configure Command => %s
 Server API => Command Line Interface
 Virtual Directory Support => %s
 Configuration File (php.ini) Path => %s
