Index: php-5.2.17/ext/sqlite/config.m4
===================================================================
--- php-5.2.17.orig/ext/sqlite/config.m4	2015-04-13 04:58:09.269313480 +0200
+++ php-5.2.17/ext/sqlite/config.m4	2015-04-13 04:58:46.313312734 +0200
@@ -84,7 +84,7 @@
     ])
     SQLITE_MODULE_TYPE=external
     PHP_SQLITE_CFLAGS=$pdo_inc_path
-    sqlite_extra_sources="libsqlite/src/encode.c"
+    sqlite_extra_sources=""
   else
     # use bundled library
     PHP_PROG_LEMON
