Index: php-5.2.17/ext/reflection/config.m4
===================================================================
--- php-5.2.17.orig/ext/reflection/config.m4	2015-04-13 04:58:09.225313483 +0200
+++ php-5.2.17/ext/reflection/config.m4	2015-04-13 04:58:46.525312730 +0200
@@ -2,7 +2,7 @@
 dnl config.m4 for extension reflection
 
 PHP_ARG_ENABLE(reflection, whether to enable reflection support,
-[  --disable-reflection    Disable reflection support], yes, no)
+[  --disable-reflection    Disable reflection support], yes)
 
 if test "$PHP_REFLECTION" != "no"; then
   AC_DEFINE(HAVE_REFLECTION, 1, [Whether Reflection is enabled])
