Index: php-5.2.17/ext/standard/tests/general_functions/phpinfo.phpt
===================================================================
--- php-5.2.17.orig/ext/standard/tests/general_functions/phpinfo.phpt	2015-04-13 04:58:45.677312746 +0200
+++ php-5.2.17/ext/standard/tests/general_functions/phpinfo.phpt	2015-04-13 04:58:47.777312705 +0200
@@ -1,5 +1,7 @@
 --TEST--
 phpinfo()
+--SKIPIF--
+<?php die("SKIP phpinfo - test suite's handling of "%s" is incompatible with this test"); ?>
 --FILE--
 <?php
 var_dump(phpinfo());
@@ -24,6 +26,7 @@
 Virtual Directory Support => %s
 Configuration File (php.ini) Path => %s
 Loaded Configuration File => %a
+Scan this dir for additional .ini files => %s
 PHP API => %d
 PHP Extension => %d
 Zend Extension => %d
