From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Fri, 20 Mar 2015 12:08:45 +0100
Subject: 100-recode_is_shared

---
 ext/recode/config9.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/ext/recode/config9.m4
+++ b/ext/recode/config9.m4
@@ -13,6 +13,6 @@ if test "$PHP_RECODE" != "no"; then
   fi
 
   if test -n "$recode_conflict"; then
-    AC_MSG_ERROR([recode extension can not be configured together with:$recode_conflict])
+    AC_MSG_WARN([recode extension can not be used together with:$recode_conflict])
   fi
 fi
