From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Fri, 20 Mar 2015 12:08:46 +0100
Subject: qdbm-is-usr_include_qdbm

---
 ext/dba/config.m4 | 4 ++++
 1 file changed, 4 insertions(+)

--- a/ext/dba/config.m4
+++ b/ext/dba/config.m4
@@ -113,6 +113,10 @@ if test "$PHP_QDBM" != "no"; then
       THIS_PREFIX=$i
       THIS_INCLUDE=$i/include/depot.h
       break
+    elif test -f "$i/include/qdbm/depot.h"; then
+      THIS_PREFIX=$i
+      THIS_INCLUDE=$i/include/qdbm/depot.h
+      break
     fi
   done
 
