From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Fri, 20 Mar 2015 12:08:47 +0100
Subject: hurd-noptrace

---
 sapi/fpm/config.m4 | 5 +++++
 1 file changed, 5 insertions(+)

--- a/sapi/fpm/config.m4
+++ b/sapi/fpm/config.m4
@@ -147,6 +147,11 @@ AC_DEFUN([AC_FPM_TRACE],
         pid_t child;
         int status;
 
+        /* broken ptrace on Hurd, avoid hanging */
+        #ifdef __GNU__
+        return 10;
+        #endif
+
         if ( (child = fork()) ) { /* parent */
           int ret = 0;
 
