--- a/build/buildcheck.sh
+++ b/build/buildcheck.sh
@@ -44,17 +44,17 @@ echo "           to build PHP from SVN."
 exit 1
 fi
 
-if test "$1" = "2" -a "$2" -gt "59"; then
-  echo "buildconf: You need autoconf 2.59 or lower to build this version of PHP."
-  echo "           You are currently trying to use $ac_version"
-  echo "           Most distros have separate autoconf 2.13 or 2.59 packages."
-  echo "           On Debian/Ubuntu both autoconf2.13 and autoconf2.59 packages exist."
-  echo "           Install autoconf2.13 and set the PHP_AUTOCONF env var to "
-  echo "           autoconf2.13 and try again."
-  exit 1
-else
+#if test "$1" = "2" -a "$2" -gt "59"; then
+#  echo "buildconf: You need autoconf 2.59 or lower to build this version of PHP."
+#  echo "           You are currently trying to use $ac_version"
+#  echo "           Most distros have separate autoconf 2.13 or 2.59 packages."
+#  echo "           On Debian/Ubuntu both autoconf2.13 and autoconf2.59 packages exist."
+#  echo "           Install autoconf2.13 and set the PHP_AUTOCONF env var to "
+#  echo "           autoconf2.13 and try again."
+#  exit 1
+#else
   echo "buildconf: autoconf version $ac_version (ok)"
-fi
+#fi
 
 if test "$1" = "2" -a "$2" -ge "50"; then
   ./vcsclean
